charts/victoria-metrics-agent/values.yaml

```diff
@@ -4,6 +4,24 @@
 
 replicaCount: 1
 
+## Deployment or StatefulSet
+# Deployment ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/
+# StatefulSet ref:  https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/
+kind: Deployment
```

Comment:
actually, what do you think about  defining the top level properties
statefulset:
 enabled: bool
 fileds related to sts
deployemnt:
 enabled: bool
  ....
?

Reply:
it could be done like this, and any issue I don't see. I'll make the commit

Only, I don't like the PVC https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-agent/templates/pvc.yaml because it can confuse people, between PVC template in statefulset and PVC in deployment but if it removed, the users faced the issue, they should switch from deployment to statefulset,and remove old PVC who is using chart now. Maybe do you have an idea ?

Reply:
thank you, as sts a new entity we can define  own pvc inside an object

Reply:
> actually, what do you think about defining the top level properties
> 
> ```yaml
> statefulset:
>  enabled: bool
>  fileds related to sts
> deployemnt:
>  enabled: bool
>   ....
> ```
> 
> ?

Are we able to enabled statefulset and deployment at once?
or if statefulset is enabled then deployment should be disabled automaticly?

Reply:
So, it could be enabled deployment and statefulset at one moment =) I guess users will see the issue if they  are enabled this two option at once